src/sagemaker/estimator.py

```diff
@@ -1416,6 +1416,7 @@ def __init__(
         checkpoint_s3_uri=None,
         checkpoint_local_path=None,
         enable_sagemaker_metrics=None,
+        s3_client=None,
```

Comment:
given that one can already pass in a sagemaker session to an estimator, it seems like it would be more appropriate to configure a sagemaker session with the desired s3 client rather than introducing a slightly overlapping parameter here.

Reply:
@laurenyu - thank you for the feedback. I will check if this is feasible!

Reply:
@laurenyu - I have refactored now to allow to pass in an s3_endpoint_url; this then creates the s3_resource and s3_client objects. Wherever these are used it will check if one is available, or create one. All parts of the SDK should now benefit from it. The feature is only available in local mode.